### ISSUE TITLE
DeleteScope fails if it's linked to a parent

### ISSUE THREAD

**Issue by sandorfr** [geeklearningio/Bouncer#41]
Currently, when deleting a scope it deletes the child hierarchy but it fails to remove the parent links leading to an integrity violation.

```
The DELETE statement conflicted with the REFERENCE constraint "FK_ScopeHierarchy_ChildScope". 
The conflict occurred in database "workbench-guid", table "Authorizations.ScopeHierarchy", column 'ChildId'.
```

**Comment by sandorfr** [comment]
It seems that the in memory db context does not to integrity constraints check which makes it a bit harder to test

**Comment by sandorfr** [comment]
I have a PR pending https://github.com/geeklearningio/gl-dotnet-authorizations/pull/42 but the test doesn't test anything, I'd like to get your input on that one. 

In addition, I update it it so it removes some redundant name lookups and also uses the primary key whenever possible (which is the Clustered faster Index).

**Comment by sandorfr** [comment]
I noticed it's not removing related authorizations. I would have expected it to do it as well.

**Comment by arnaudauroux** [comment]
Fixed in 0.2.0
